src/Model/Scope/BasicCondition.php

```diff
@@ -0,0 +1,327 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\data\Model\Scope;
+
+use atk4\core\ReadableCaptionTrait;
+use atk4\data\Exception;
+use atk4\data\Field;
+use atk4\dsql\Expression;
+use atk4\dsql\Expressionable;
+
+class BasicCondition extends AbstractCondition
+{
+    use ReadableCaptionTrait;
+
+    public $key;
+
+    public $operator;
+
+    public $value;
+
+    protected static $opposites = [
+        '=' => '!=',
+        '!=' => '=',
+        '<' => '>=',
+        '>' => '<=',
+        '>=' => '<',
+        '<=' => '>',
+        'LIKE' => 'NOT LIKE',
+        'NOT LIKE' => 'LIKE',
+        'IN' => 'NOT IN',
+        'NOT IN' => 'IN',
+        'REGEXP' => 'NOT REGEXP',
+        'NOT REGEXP' => 'REGEXP',
+    ];
+
+    protected static $dictionary = [
+        '=' => 'is equal to',
+        '!=' => 'is not equal to',
+        '<' => 'is smaller than',
+        '>' => 'is greater than',
+        '>=' => 'is greater or equal to',
+        '<=' => 'is smaller or equal to',
+        'LIKE' => 'is like',
+        'NOT LIKE' => 'is not like',
+        'IN' => 'is one of',
+        'NOT IN' => 'is not one of',
+        'REGEXP' => 'is regular expression',
+        'NOT REGEXP' => 'is not regular expression',
+    ];
+
+    protected static $skipValueTypecast = [
+        'LIKE',
+        'NOT LIKE',
+        'REGEXP',
+        'NOT REGEXP',
+    ];
+
+    public function __construct($key, $operator = null, $value = null)
+    {
+        if (func_num_args() == 2) {
+            $value = $operator;
+            $operator = '=';
+        }
+
+        if (is_bool($key)) {
```

Comment:
usecase?

Reply:
`BasicCondition(false)` basically sets all records off-limit and is used in record access control.

Reply:
I understard, there is no enable/disable now, right?

I do not like constructor that accepts "for one usecase" absolutelly different pattern/type.

what are currently other regular alternatives without allowing bool to enable/disable the condition?

as this is in constructor anyway, we can have static method like `alwaysFalse()` which can produce the same result

Reply:
It is not about enabling/disabling a condition. This is no longer available as option.
`BasicCondition(false)` translates to `BasicCondition(Expression('false'))`. Any expression is a legitimate condition.
In SQL terms this particular case is transformed to `SELECT * FROM xyz WHERE false AND ...`.
`BasicCondition(true)` translates to `BasicCondition()` or simply an empty condition (which is skipped when generating SQL).
It makes no sense to introduce special class or method for it, etc. This just adds unnecessary complexity.

I completely removed the activate/deactivate functionality. This is covered by the `clear` method now. 
The use case is related to placeholders and `clear` method is enough.

Reply:
> I do not like constructor that accepts "for one usecase" absolutelly different pattern/type.

this is what should be solved - true/false should not be allowed

Reply:
What is the problem with some flexibility?
You know we are on different opinions about flexibility 😉

Reply:
Flexibility is very dangerous and high degree of freedoms is very hard to test.

Imagine code like this:
```
function getField() {
if (not loaded) {
return false;
}
}

BasicCondition(getField());
```

so if bool is not supported, you will get an error, otherwise it may take a long time to find this and flexibility equals to nighmare

so lets make a compromise - test for `func_num_args() == 1` and drop this functionality in the future if it will became common source of issues

Reply:
Agree